### PR TITLE
Use new API_ROOT setting to build absolute URLs

### PR DESCRIPTION
**Pull request by kdmccormick**
@edx/masters-neem 
[EDUCATOR-4274](https://openedx.atlassian.net/browse/EDUCATOR-4274)

Configuration PR: https://github.com/edx/configuration/pull/5102
Internal configuration PR: https://github.com/edx/edx-internal/pull/849

### CODE REVIEW

File: registrar/apps/api/utils.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    """
+    Convert a path to a resource on Registrar to an absolute,
+    externally-accessible URL.
+
+    Arguments:
+        path (str)
+
+    Returns: str
+    """
+    return settings.URL_ROOT.strip('/') + '/' + path.strip('/')
```
Comment (iloveagent57) [inline registrar/apps/api/utils.py]: Nit: you could make this a little more precise with `rstrip` and `lstrip`:
```python
return settings.URL_ROOT.rstrip('/') + '/' + path.lstrip('/')
```
That way you don't have to worry about stripping off a trailing "/" (I don't know if that's even a consideration, here, though).

File: registrar/apps/api/utils.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    absolute, externally-accessible URL.
+
+    Arguments:
+        url_name (str): name of URL as configured in urls.py.
+        kwargs (dict): URL kwargs for call to `reverse`.
+
+    Returns: str
+    """
+    raw_url = reverse(url_name, kwargs=kwargs)
+    if raw_url.startswith('/api/'):
```
Comment (iloveagent57) [inline registrar/apps/api/utils.py]: NIT: you know that `raw_url` starts with "/api/" inside this block, so why not do:
```python
return settings.API_ROOT.rstrip('/') + '/' + raw_url.lstrip('/api/')
```

**Review by iloveagent57 — COMMENTED** [review]
I know this is a draft, just a couple minor suggestions about string munging.

File: registrar/apps/api/utils.py (hunk 1 above)
Comment (kdmccormick) [reply registrar/apps/api/utils.py]: Ended up rewriting this part, but I made sure to preserve the presence of a trailing slash this time.

File: registrar/apps/api/utils.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+            => 'https://{API_ROOT}/api/v1/programs/123'
+
+    Returns: str
+    """
+    PREFIX = '/api/'
+    if not path.startswith(PREFIX):
+        raise ValueError(
+            'Cannot make API URL for raw URL that does not begin with ' + PREFIX
+        )
+    path_parts = [settings.API_ROOT, path[len(PREFIX):]] + list(more_paths)
```
Comment (schenedx) [inline registrar/apps/api/utils.py:47]: Please write a unit test for this block of logic. It's difficult to access the different case you are trying to handle by reading this code alone.

**Review by schenedx — COMMENTED** [review]
I think we just need a set of unit test, and this is good to go.

**Comment by zacharis278** [comment]
lgtm pending some tests.  However, if we're using /media to serve up the results there's nothing exposing that path in api.yaml.

**Review by zacharis278 — APPROVED** [review]

**Pull request merged**